torchtune/training/_compile.py

```diff
@@ -86,3 +86,10 @@ def compile_loss(loss: nn.Module, verbose: bool = True) -> nn.Module:
     else:
         loss = torch.compile(loss, backend=backend)
     return loss
+
+
+def compile_optimizer_step(optimizer_step_fn, verbose: bool = True):
```

Comment:
I appreciate you wanting to keep this similar to how we were currently doing things; however, we only needed to this for the loss function b/c we were doing funky things with chunking. 

We should just compile this directly in the recipe. Same goes for the other PR you have up.